tools/rosbag_storage/src/buffer.cpp

```diff
@@ -65,7 +66,12 @@ void Buffer::ensureCapacity(uint32_t capacity) {
         capacity_ = capacity;
     else {
         while (capacity_ < capacity)
+        {
+          if (static_cast<uint64_t>(capacity) * 2 >=  std::numeric_limits<uint32_t>::max())
+            capacity_ = capacity;
```

Comment:
Why should this be set to `capacity` and not to `std::numeric_limits<uint32_t>::max()` since the general policy is to grow by a factor of 2? Otherwise this would reallocate every time the capacity is increases by e.g. 1 if it is already greater than `max / 2`.

Reply:
Good idea.